crates/core/tedge_agent/src/operation_workflows/persist.rs

```diff
@@ -264,18 +263,15 @@ impl WorkflowRepository {
         operation: &OperationName,
         version: &WorkflowVersion,
     ) {
-        if let Some(in_use_version) = self.in_use_copies.get(operation) {
-            if in_use_version == version {
-                return;
-            }
+        if self.in_use_copies.contains(version) {
+            return;
         };
         if let Some((_, source)) = self.definitions.get(operation) {
-            let target = self.state_dir.join(operation).with_extension("toml");
+            let target = self.state_dir.join(version).with_extension("toml");
```

Comment:
How about using a combination of `operation` and `version` in the file name so that we're not looking at a random hash number during debugging?

Reply:
A `grep` should make it, but why not.

Reply:
Done along https://github.com/thin-edge/thin-edge.io/pull/3180/commits/ae56d97652643352a7e94e2b9ee73b93da3e6df6